examples/memory_estimation.ipynb

```diff
@@ -0,0 +1,218 @@
+{
```

Comment:
Nice! I have tried to run it on my laptop, I also see the issue with the first run being a bit "off". I would actually keep a single set of matrix sizes/cluster parameters (right now the first set of parameters is repeated twice), and add one line suggesting to run the cell twice if the result for the first set of parameters is a bit off, probably due to some initialisations within memory_profiler.. 

Minor point: I have also noticed the `sys` package is used but not imported.